gnucash/gnome/window-reconcile.c

```diff
@@ -1365,11 +1421,50 @@ gnc_ui_reconcile_window_delete_cb(GtkButton *button, gpointer data)
             return;
     }
 
+    /* Save the view and split to be selected after split is deleted */
+    save_view = GNC_RECONCILE_VIEW (recnData->debit);
+    selection = gtk_tree_view_get_selection (GTK_TREE_VIEW (save_view));
+    if (gtk_tree_selection_count_selected_rows(selection) == 0)
+    {
+        save_view = GNC_RECONCILE_VIEW (recnData->credit);
+        selection = gtk_tree_view_get_selection (GTK_TREE_VIEW (save_view));
+    }
```

Comment:
This took me a while to understand the above, as the delete button is used for credit and debits you are deciding what view to use. I would suggest adding a comment here or add a function to return the view, you could probably base it on  gnc_reconcile_window_get_current_split and return a view instead of a split.

Reply:
Fixed in PR 707 (against master). There is new function (gnc_reconcile_window_get_selection_view) for getting the view containing the selection with Doxygen comment.